### PR TITLE
FHT: Hosting trial appears in the plans-new-hosted-site plans grid

### PR DESCRIPTION
**Pull request by p-jackson**
Closes https://github.com/Automattic/dotcom-forge/issues/4008.

## Proposed Changes

* Adds the free trial to the plans grid during the `/setup/new-hosted-site/plans` step
* Plan gets added to the site and transferred to Atomic
* User is redirected to `/home/%s`

image

## Testing Instructions

* Confirm that we aren't inadvertently adding the trial plan to the plans grid in any unexpected locations! e.g. the trial plan appearing during the default `/start` flow.

## Known issues

- there is no `$0` (this should be displayed after we add the trial plan to the `/plans` endpoint)
- the Atomic transfer is not yet happening, this will be addressed in https://github.com/Automattic/dotcom-forge/issues/4046#top. The network request says that no Atomic request was initiated

### CODE REVIEW

**Comment by vykes-mac** [comment]
Coming along nice, code looks good! Plan didn't show up where it's not supposed to.

File: packages/calypso-products/src/plans-list.tsx [hunk 1]
```diff
@@ -3622,11 +3623,11 @@ if ( isEnabled( 'plans/migration-trial' ) ) {
 
 PLANS_LIST[ PLAN_HOSTING_TRIAL_MONTHLY ] = {
 	...getPlanBusinessDetails(),
-	type: TYPE_BUSINESS,
+	type: TYPE_HOSTING_TRIAL,
 	group: GROUP_WPCOM,
 	getProductId: () => 1058,
 	getPathSlug: () => PLAN_HOSTING_TRIAL_MONTHLY,
-	term: TERM_MONTHLY,
+	term: TERM_ANNUALLY,
```
Comment (zaguiini) [inline packages/calypso-products/src/plans-list.tsx]: Was this change intentional, @p-jackson? There is no tenure associated with the FT plan, so does it matter?

Comment (zaguiini) [reply packages/calypso-products/src/plans-list.tsx]: Oops, just saw that: "The trial plan had to be set to an annual term because the plans grid crashes when there's no annual term for a plan. I think there's a hard coded assumption that all plans must have an annual variant at least. But that's not true for us. We should see if we can fix this underlying assumption."

Please ignore!

Comment (p-jackson) [reply packages/calypso-products/src/plans-list.tsx]: > The trial plan had to be set to an annual term because the plans grid crashes when there's no annual term for a plan. I think there's a hard coded assumption that all plans must have an annual variant at least. But that's not true for us. We should see if we can fix this underlying assumption.

In the database it's a monthly plan (even though we manually adjust the expiry date to make it a week), so I feel like it'd be more consistent to have the term be `TERM_MONTHLY` in Calypso too. So that database definition matches the Calypso definition.

But yeah it's a hack to get things going. Maybe you can find a way to fix the plans grid to support products with no annual term.

File: client/my-sites/plans-grid/components/features-grid/index.tsx [hunk 2]
```diff
@@ -436,7 +438,7 @@ class FeaturesGrid extends Component< FeaturesGridType > {
 				className="plan-features-2023-grid__table-item"
 				isTableCell={ options?.isTableCell }
 			>
-				{ ! isFreePlan( planSlug ) && (
+				{ ! isFreePlan( planSlug ) && ! isFreeHostingTrial( planSlug ) && (
```
Comment (p-jackson) [inline client/my-sites/plans-grid/components/features-grid/index.tsx:441]: It'd be nice if the plan objects in the `PLANS_LIST` array had some sort of `isFree` prop so we could handle this branching for _all_ free plans. Like free Jetpack too for example.

File: client/my-sites/plans-grid/components/billing-timeframe.tsx [hunk 3]
```diff
@@ -23,7 +24,11 @@ function usePerMonthDescription( { planSlug }: { planSlug: PlanSlug } ) {
 		storageAddOnsForPlan,
 	} = gridPlansIndex[ planSlug ];
 
-	if ( isWpComFreePlan( planSlug ) || isWpcomEnterpriseGridPlan( planSlug ) ) {
+	if (
+		isWpComFreePlan( planSlug ) ||
+		isWpcomEnterpriseGridPlan( planSlug ) ||
+		planSlug === PLAN_HOSTING_TRIAL_MONTHLY
+	) {
```
Comment (p-jackson) [inline client/my-sites/plans-grid/components/billing-timeframe.tsx:31]: Ah I think this is what I needed to make the `TERM_MONTHLY` thing work for the trial plan! 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx [hunk 4]
```diff
@@ -159,6 +163,16 @@ const SiteCreationStep: Step = function SiteCreationStep( { navigation, flow, da
 			sourceSlug
 		);
 
+		if ( planCartItem?.product_slug === PLAN_HOSTING_TRIAL_MONTHLY && site ) {
+			await addHostingTrial( { siteId: site.siteId, planSlug: PLAN_HOSTING_TRIAL_MONTHLY } );
```
Comment (p-jackson) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx:167]: 😎 

Great! This is tidier than I was imagining it would need to be.

**Review by p-jackson — COMMENTED** [review]
Great!

Tested and clicking the trial button creates the site and starts the polling for the atomic transfer to complete (which is obviously still a WIP).

Great starting point :shipit:

**Comment by p-jackson** [comment]
✅

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9225140

Thank you @p-jackson for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9225140).
